sunpy/net/dataretriever/client.py

```diff
@@ -264,35 +173,27 @@ def _get_full_filenames(self, qres, filenames, path):
 
         return paths
 
-    def _get_time_for_url(self, urls):
-        """
-        This method allows clients to customise the timerange displayed for
-        each URL.
-
-        It should return a sunpy.time.TimeRange object per URL.
-        """
-        return NotImplemented
-
     def search(self, *args, **kwargs):
         """
         Query this client for a list of results.
-
-        Parameters
-        ----------
-        \\*args: `tuple`
-            `sunpy.net.attrs` objects representing the query.
         """
-        GenericClient._makeargs(self, *args, **kwargs)
-
-        kwergs = copy.copy(self.map_)
-        kwergs.update(kwargs)
-        urls = self._get_url_for_timerange(
-            self.map_.get('TimeRange'), **kwergs)
-        if urls:
-            times = self._get_time_for_url(urls)
-            if times and times is not NotImplemented:
-                return QueryResponse.create(self.map_, urls, times, client=self)
-        return QueryResponse.create(self.map_, urls, client=self)
+        baseurl, pattern, matchdict = self.pre_hook(*args, **kwargs)
+        scraper = Scraper(baseurl, regex=True)
+        for elem in args:
+            if isinstance(elem, Time):
+                timerange = TimeRange(elem.start, elem.end)
+            elif hasattr(elem, 'value'):
+                matchdict[elem.__class__.__name__] = [str(elem.value)]
+            else:
+                raise ValueError("GenericClient can not add {} to the map_ dictionary to pass to the Client.".format(elem.__class__.__name__))
+        for k in kwargs:
+            matchdict[k] = [str(kwargs[k])]
+        filesmeta = scraper._extract_files_meta(timerange, extractor=pattern, matcher=matchdict)
+        metalist = []
+        for i in filesmeta:
+            map_ = self.post_hook(i, matchdict)
```

Comment:
What do you think the pros/cons of putting this loop inside `post_hook` are? Are there cases where the source would want to handle all the returns as a group? i.e. to filter them or something mad?

Reply:
Passing large `filesmeta`list of dictionaries which may contain a lot of metada to a method will increase memory usage (I'm not very certain about it).
There are no filtering left, all that is handled by the `matchdict` passed in scraper, as we decided in the last meeting.
The `post_hook` will be overwritten for clients like `norh` to convert `tcz` or `tca` in the url to `17*u.GHz` or `34 *u.GHz`.